lib/extensions/attachable/active_record/base.rb

```diff
@@ -1,12 +1,45 @@
 module Extensions::Attachable::ActiveRecord::Base
   module ClassMethods
     # This function should be declared in model, to it have attachments.
-    def acts_as_attachable
-      has_many :attachments, as: :attachable, inverse_of: :attachable
+    def has_many_attachments # rubocop:disable Style/PredicateName
+      has_many :attachments, as: :attachable, inverse_of: :attachable,
+                             dependent: :destroy, autosave: true
 
-      accepts_nested_attributes_for :attachments,
-                                    allow_destroy: true,
-                                    reject_if: -> (params) { params[:attachment].blank? }
+      define_method(:files=) do |files|
+        files.each do |file|
+          attachments.build.file_upload = file
+        end
+      end
+    end
+
+    def has_one_attachment # rubocop:disable Style/PredicateName
```

Comment:
I've the same concern before, but `files=` is not needed in `has_one_attachment`. So I came with this implementation,  actually they only share the association. What do you think ?